integration/test/crmapping/crmapping_test.go

```diff
@@ -0,0 +1,132 @@
+package crmapping
+
+import (
+	"context"
+	"io/ioutil"
+	"path"
+	"strconv"
+	"testing"
+
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/giantswarm/operatorkit/resource"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/giantswarm/azure-operator/v4/service/controller/resource/capzcrs"
+)
+
+func TestAzureConfigCRMapping(t *testing.T) {
+	testCases := []struct {
+		name         string
+		inputFiles   []string
+		errorMatcher func(error) bool
+	}{
+		{
+			name: "case 0: Simple AzureConfig",
+			inputFiles: []string{
+				"simple_azureconfig.yaml",
+			},
+			errorMatcher: nil,
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			t.Log(tc.name)
+
+			client := newFakeClient()
+			ensureCRsExist(t, client, tc.inputFiles)
+
+			r := constructResource(t, client)
+
+			err := r.EnsureCreated(context.Background(), nil)
+
+			// TODO: Read expected CRs w/ client and compare to "golden" versions.
```

Comment:
In next PR I start the implementation of `capzcrs` functionality. Basic logic is required there in order to being able to then implement the gold file structure here. I'll provide both in the same PR then.

My idea here is to have set of input files that are "loaded" into fake client (i.e. the baseline before executing the handler) and then run the mapping once and then compare the resulting state with what is stored in "golden files".

This way it's easy to make thorough test coverage for CR fields while keeping maintenance work low when changing the mapping. One should then also see nicely the concrete diff of changes in end result CRs when changing the mapping logic.